### PR TITLE
B/wep 174 fix focus empty rtb firefox

### PR DESCRIPTION
**Pull request by faebebin**
Update: thanks for your reviews. @roosnic1 's idea to implement a helper as an editor method made me add quite some changes to stay consistent. This new commit is only refactoring and should not change any behaviour you reviewed already. I tried to stick to the Slate docs with implementing this change.

The problem with re-focusing an empty richtextblock, which only occured on Firefox, was related to the placeholder of an empty block. The problem seems popular on the web and not nicely solved. Hopefully there will be a solution in future releases. For now the workaround is just removing the builtin placeholder. I also added a little alternative placeholder div. The placeholder could be left aside completely (blank) if it is more disturbing than helping in your opinions. The Firefox users, please really try the new behaviour and check if improved or some problems presist.

### CODE REVIEW

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx [hunk 1]
```diff
@@ -94,13 +98,21 @@ export const RichTextBlock = memo(function RichTextBlock({
...
       <Editable
         readOnly={disabled}
-        placeholder={t('blocks.richText.startWriting')}
+        // placeholder={t('blocks.richText.startWriting')}  # causes focusing problems on firefox !
         renderElement={renderElement}
         renderLeaf={renderLeaf}
         onBlur={() => {
           setLocation(editor.selection)
+          setIsEmpty(
+            JSON.stringify(Array.from(editor.children)) === JSON.stringify(createDefaultValue())
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx:114]: maybe move this to a function inside the editor? `editor.isEmpty()` which returns a boolean. Might be useful in other places.

**Review by bishoymelek — DISMISSED** [review]

**Review by 0815pascal — APPROVED** [review]

**Review by penk85 — APPROVED** [review]
Looks good and works smoothly in Firefox.

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx (hunk 1 above)
Comment (faebebin) [reply packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx:114]: Good idea. Will update a suggestion according to https://docs.slatejs.org/concepts/07-plugins#helper-functions .

**Review by roosnic1 — APPROVED** [review]
👍🏾

**Review by penk85 — APPROVED** [review]

**Pull request merged**